### PR TITLE
added support for cifar10 and automatic resolution selection in training example

### PR DESCRIPTION
**Pull request by lukovnikov**
Added support for CIFAR10 in training example and automatic resolution selection + printing of param counts

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
@anton-l could you take a look here?

File: examples/unconditional_image_generation/train_unconditional.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def parse_args():
     parser.add_argument(
         "--resolution",
         type=int,
-        default=64,
+        default=-1,
```
Comment (patrickvonplaten) [inline examples/unconditional_image_generation/train_unconditional.py:95]: Not a fan of changing the default here - examples are meant to be easily readable. A default resolution of -1 is impossible to understand for people

File: examples/unconditional_image_generation/train_unconditional.py [hunk 2]
```diff
@@ -253,6 +263,9 @@ def main(args):
             "UpBlock2D",
         ),
     )
+
+    print(f"Number of parameters: {count_parameters(model)//1e6:.2f}M")
```
Comment (patrickvonplaten) [inline examples/unconditional_image_generation/train_unconditional.py:267]:
```suggestion
    logger.info(f"Number of parameters: {count_parameters(model)//1e6:.2f}M")
```

**Review by patrickvonplaten — APPROVED** [review]
This generally looks good to me :-) Could you leave the default to 64 though? And then maybe we'll just add a section in the README on how to run it for CIFAR? 

Also cc @anton-l would be nice if you could take a look :-)

**Comment by lukovnikov** [comment]
> This generally looks good to me :-) Could you leave the default to 64 though? And then maybe we'll just add a section in the README on how to run it for CIFAR?

What about changing default to "None" instead and changing the `help` description field so it's clear that the script chooses resolution based on dataset automatically? If not, I'll put 64 back :), I was just trying to unbloat the command line usage from redundant info.

**Comment by anton-l** [comment]
Let's add 64 back indeed, this way the choice of resolution will be explicit and controllable (e.g. for resolutions of more than 64, not only CIFAR and the existing examples)

**Pull request closed without merging**